extensions/network-icon.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Network Icon"
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Simon Ser"
+    period: "2025"
+    email: "contact@emersion.fr"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+## Description
+
+The `NETWORK` ISUPPORT token allows servers to advertise their network name. This specification adds a new ISUPPORT token to advertise a network icon.
+
+## The `draft/NETWORKICON` ISUPPORT token
+
+If a server publishes a `draft/NETWORKICON` ISUPPORT token, the value MUST be a URL to an image. The URL SHOULD use the "https" scheme.
```

Comment:
should there be a way for the client to say what mime types it accepts for the image to be in?

Reply:
MIME types are one thing, image size in pixels is another one… For the latter negotiation wouldn't work, the client might want to show the icon in different places, with a different size for each one.

The Web solves this by listing all possible icons, with all formats and sizes. I've decided against this approach for this spec because:

- It adds a whole bunch of extra complexity.
- It's a lot of characters. URLs might be quite long.

If we really wanted to solve this, maybe advertising the URL of a JSON manifest would be better (like [Web app manifests](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Manifest)). But I don't think we actually need this complexity.

Reply:
Something like `.ico` that can pack multiple sizes could work in this regard.

Reply:
`.ico` is a very old format, I don't think we should build on it. It's missing a lot of features one would expect from modern image formats.

Reply:
one option, to move the complexity elsewhere, would be to only support svg icons.

Reply:
My expectation is that server authors will use one of the widely supported image formats (e.g. PNG or SVG).

If format negotiation is really deemed necessary by a network, HTTP supports this feature natively already via [Accept](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Accept).

Reply:
I don't think this spec should leave the format entirely open-ended. If the spec allows any format, then are clients also expected to support each and every single format that could be rasterized to an image? SVG, PDF, PS, Xfig?
If we go for only HTTP icon location (which I'm not a big fan of, but anyway), then the `Accept` header should probably be mentioned explicitly so that clients and servers are advised to use it. And the server be required to provide **at least** one of (for instance) ICO, BMP, PNG. So that lightweight clients could support only these 3 formats and be assured to get the network icon, if any.

Also, FWIW, ICO format does support PNG as individual images and support alpha channel for both BMP and PNG individual images.

Reply:
+1 for using the Accept header. The server can respond with 406 if its not available in that form.